### PR TITLE
Feat/IUT-11/IconButton Component

### PR DESCRIPTION
**Pull request by neli-rashkova**
## Description
Created a reusable `IconButton` component with size variants: 'tiny',  'small', 'medium', 'large'. Size determines the padding values for the button and the width and height values for the icon. Props to pass are:

- size: a string that includes any of the strings mentioned above for size
- icon: the name of the icon (ie. MagnifyingGlassIcon)
- disabled: a boolean value (default value is false)

## Testing Instructions

1. Checkout `feat/IUT-11/IconButton` branch
2. `npm run storybook`

## Storybook Screenshots
Screenshot 2024-02-03 at 4 10 26 PM
Screenshot 2024-02-03 at 4 10 48 PM
Screenshot 2024-02-03 at 4 12 04 PM

### CODE REVIEW

File: src/components/IconButton/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import * as React from "react"
+import { Slot } from "@radix-ui/react-slot"
+import { cva, type VariantProps } from "class-variance-authority"
+
+import { cn } from "../../../lib/utils"
+
+const buttonVariants = cva(
+  "border-box inline-flex flex-column items-center justify-center gap-2 rounded border border-solid border-stroke-muted bg-bg-color-default shadow-shadow-disabled relative ease-in duration-250 hover:bg-bg-color-brand-disabled active:bg-bg-color-brand-disabled active:shadow-inner-pressed-muted focus:outline-none focus:border-2 focus:border-stroke-focus focus:shadow-none disabled:pointer-events-none disabled:bg-bg-color-disabled disabled:border-stroke-muted disabled:shadow-none",
```
Comment (joseph-hc-jun) [inline src/components/IconButton/Button/Button.tsx:8]: nit: let's remove `border-box` if it doesn't affect the styling. Just to reduce the number of classes a bit 😄

**Review by joseph-hc-jun — APPROVED** [review]
Well done @neli-rashkova ! I've left a small nit comment but feel free to merge in after addressing it. Thank you!

File: src/components/IconButton/Button/Button.tsx (hunk 1 above)
Comment (neli-rashkova) [reply src/components/IconButton/Button/Button.tsx]: Done!

**Review by launchupio — APPROVED** [review]

**Comment by launchupio** [comment]
Great job @neli-rashkova I think your merge was blocked because the requested changes forces codeowner to approve perhaps.

**Pull request merged**